src/ip_blacklist_manager.py

```diff
@@ -0,0 +1,103 @@
+import subprocess
+import argparse
+
+class BlackListManager:
+    """IpLinux is responsible for managing a list of IP addresses to be blacklisted on a linux remote server firewall
+    This class provides functionalities to:
+
+    1. Create an IP List
+    2. Check if IP list exists
+    3. Add an IP to the IP list
+    4. Add a rule blocking the blacklist
+    5. Displaying all results"""
+
+    iplist = 'dp_blacklist'
+
+    def __init__(self, ip_list_file):
+        self.ip_list_file = ip_list_file
+
+    def list_exists(self):
+        result = subprocess.run(['sudo', 'ipset', 'list', self.iplist], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+        if "Name:" in result.stdout:
+            print("Set can't be created")
+            return False
+        else:
+            print("Set can be created")
+            return True
+
+    def create_iplist(self):
+        if self.list_exists():
+            subprocess.run(['sudo', 'ipset', 'create', self.iplist, 'hash:ip'])
+            print(f"{self.iplist} has been created in the ipset")
+        return
+
+
+    def add_ip_ipset(self):
+        first_result = subprocess.run(['sudo', 'ipset', 'list', self.iplist], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+        for line in first_result.stdout.splitlines():
+            if line.startswith('Number of entries:'):
+                num_entries = line.split(':')[1].strip()
+                print(num_entries + " ips are in this list initially")
+
+        with open(self.ip_list_file) as file:
+            for ip in file:
+                stripped_ip = ip.strip()
```

Comment:
- check if the string is valid IP(meaning x.x.x.x). if not, the value should be skipped.
- in this loop, you can count the total number of valid IP in the file so that you can display the number in display_results()